### ISSUE TITLE
Update terraform-plugin-sdk to v2

### ISSUE THREAD

**Issue by jakefhyde** [rancher/terraform-provider-rke#353]
The provider currently uses `github.com/hashicorp/terraform-plugin-sdk@v1.17.2`. v2 provides noticeable improvements, most notably the ability to debug the provider from within an IDE. There is a draft PR already up: https://github.com/rancher/terraform-provider-rke/pull/347

**Comment by jakefhyde** [comment]
This is effectively an upgrade  of the entire rke terraform provider, so this should be tested as much as possible. There are no direct test cases for this issue, but it will likely be covered via normal testing as part of the release process.

**Comment by a-blender** [comment]
@jakefhyde Customers are requesting TFP rke 1.4.2 for k8s 1.25 clusters which would be out of band from the next rancher release. Should we wait to release rke 1.4.2 with rancher q2 2.7.x so that the pipeline of normal testing as part of the release process is done?

**Comment by a-blender** [comment]
Last comment is outdated! TFP RKE [v1.4.2](https://github.com/rancher/terraform-provider-rke/releases/tag/v1.4.2) has already been released and k8s 1.25 clusters are supported. Afaik, the RKE TF provider is still functioning as expected per testing from other issues but this was not tested for q2, so it should be tested for the next provider release.

**Comment by Josh-Diamond** [comment]
Ticket #353 - Test Results - ❌ - RE-OPENED
---

With tfp-rke `v1.4.2`:

| Scenario | Test Case | Result |
|---|---|---|
| 1a. | Provision a downstream RKE cluster, with 3 nodes and individual roles - [1 `etcd`, 1 `cp`, 1 `wkr`] | :x: |
| 1b. | Using cluster from Scenario 1a, upgrade k8s version | `blocked` by failure seen in scenario 1a |
| 1c. | Using cluster from Scenario 1a + 1b, add an additional worker node to the cluster | `blocked` by failure seen in scenario 1a |
| 2a. | Provision a downstream RKE cluster, w/ 1 node, all roles | :x: |
| 2b. | Using cluster from Scenario 2a, upgrade k8s version |`blocked` by failure seen in scenario 2a |
| 2c. | Using cluster from Scenario 2a + 2b, add an additional worker node to the cluster | `blocked` by failure seen in scenario 2a |

---

Scenario 1a -
1. Using the `main.tf` provided below, provision a downstream RKE cluster using k8s `v1.25.9-rancher2-2`, w/ 3 nodes, individual roles
```terraform
terraform {
  required_providers {
    rke = {
      source = "rancher/rke"
      version = "1.4.2"
    }
  }
}

provider "rke" {
  # Configuration options
}

resource "rke_cluster" "jkeslar-cluster" {
  kubernetes_version = "v1.25.9-rancher2-2"
  nodes {
    address = "<REDACTED>"
    internal_address = "<REDACTED>"
    user    = "ubuntu"
    role    = ["etcd"]
    ssh_key = file("<REDACTED>")
  }
  nodes {
    address = "<REDACTED>"
    internal_address = "<REDACTED>"
    user    = "ubuntu"
    role    = ["controlplane"]
    ssh_key = file("<REDACTED>")
  }
  nodes {
    address = "<REDACTED>"
    internal_address = "<REDACTED>"
    user    = "ubuntu"
    role    = ["worker"]
    ssh_key = file("<REDACTED>")
  }
}
```
2. Veri-FAILED: cluster fails to provision with `failed to bring up worker plane` error

Screenshot/Error:
![image (39)](https://github.com/rancher/terraform-provider-rke/assets/46494969/73d20f15-24dc-4595-bb9c-65f4b3e04631)

---

Scenario 1b - _blocked/pending_

---

Scenario 1c - _blocked/pending_

---

Scenario 2a -
1. Using the `main.tf` provided below, provision a downstream RKE cluster using k8s `v1.25.9-rancher2-2`, w/ 1 node, all-roles
```terraform
terraform {
  required_providers {
    rke = {
      source = "rancher/rke"
      version = "1.4.2"
    }
  }
}

provider "rke" {
  # Configuration options
}

resource "rke_cluster" "rke_cluster" {
  kubernetes_version = "v1.25.9-rancher2-2"
  nodes {
    address = "<REDACTED>"
    internal_address = "<REDACTED>"
    user    = "ubuntu"
    role    = ["etcd", "controlplane", "worker"]
    ssh_key = file("<REDACTED>")
  }
}
```
2. Veri-FAILED: cluster fails to provision with `failed to bring up worker plane` error

Screenshot/Error:
![image (39)](https://github.com/rancher/terraform-provider-rke/assets/46494969/43d1c7b2-9c02-4629-8197-d9951115ba23)

---

Scenario 2b - _blocked/pending_

---

Scenario 2c - _blocked/pending_
